akka-stream/src/main/scala/akka/stream/SystemMaterializer.scala

```diff
@@ -27,9 +36,51 @@ object SystemMaterializer extends ExtensionId[SystemMaterializer] with Extension
 }
 
 final class SystemMaterializer(system: ExtendedActorSystem) extends Extension {
+  private val systemMaterializerPromise = Promise[Materializer]()
+
+  // load these here so we can share the same instance across materializer guardian and other uses
+  /**
+   * INTERNAL API
+   */
+  @InternalApi @silent("deprecated")
+  private[akka] val materializerSettings = ActorMaterializerSettings(system)
+
+  private implicit val materializerTimeout: Timeout =
+    system.settings.config.getDuration("akka.stream.materializer.creation-timeout").asScala
+
+  @InternalApi @silent("deprecated")
+  private val materializerGuardian = system.systemActorOf(
+    MaterializerGuardian
+      .props(systemMaterializerPromise, materializerSettings)
+      .withDispatcher(materializerSettings.dispatcher)
+      .withDeploy(Deploy.local),
+    "Materializers")
+
+  /**
+   * INTERNAL API
+   */
+  @InternalApi
+  private[akka] def createAdditionalSystemMaterializer(): Materializer = {
+    val started =
+      (materializerGuardian ? MaterializerGuardian.StartMaterializer).mapTo[MaterializerGuardian.MaterializerStarted]
+    Await.result(started, materializerTimeout.duration).materializer
+  }
+
+  /**
+   * INTERNAL API
+   */
+  @InternalApi
   @silent("deprecated")
-  val materializer = {
-    val settings = ActorMaterializerSettings(system)
-    ActorMaterializer.systemMaterializer(settings, "default", system)
+  private[akka] def createAdditionalLegacySystemMaterializer(
+      namePrefix: String,
+      settings: ActorMaterializerSettings): Materializer = {
+    val started =
+      (materializerGuardian ? MaterializerGuardian.LegacyStartMaterializer(namePrefix, settings))
+        .mapTo[MaterializerGuardian.MaterializerStarted]
+    Await.result(started, materializerTimeout.duration).materializer
   }
+
+  // block on async creation to make it effectively final
+  val materializer = Await.result(systemMaterializerPromise.future, materializerTimeout.duration)
```

Comment:
Since we expect the default system wide materializer to be used all over I'm avoiding having to make it volatile/atomic/mutable with this little scheme, not sure it is safe.